sdk/client/client.ts

```diff
@@ -124,39 +117,98 @@ export class Client {
   basePath: string
   // The available API endpoints
   api: Api
+
   // The path to the secrets file which may be used to store credentials
-  secretsFilePath: string
+  secretsFilePath: string = process.cwd() + '/secrets.json'
+  secretsFileContent: { api: {} }
+
   // The credential access details
   private tokenUrlDetails: [Source, string]
   private usernameDetails: [Source, string]
   private passwordDetails: [Source, string]
   private clientIdDetails: [Source, string]
   private clientSecretDetails: [Source, string]
+
   // The refresh limit in seconds before token expiry to trigger a refresh
-  refreshLimit: number = refreshLimit
+  refreshLimit: number = 3580
+
+  private loadSecretsFile(): object | null {
+
+    try
+    {
+      this.secretsFileContent = require( this.secretsFilePath );
+    }
+    catch( e )
+    {
+      // not necessarily a problem. even if the format of the source file is bad, this will reset the internal validation to an empty configuration
+
+      this.secretsFileContent = { api: {} };
+    }
```

Comment:
What are your thoughts on try/catch handling vs if/then handling? We could check that the file exists and if it does not raise a warning rather than silently defaulting to a default empty set of values. 

I'm not sure which is better suited to Typescript and more specifically Node.js